### DISCUSSION TITLE
Not all dependencies have a build. How to use ^build?

### DISCUSSION THREAD

**Discussion by shlroland** [vercel/turborepo#6842] [Help] [answered]
This is my file directory structure.

<img width="367" alt="image" src="https://github.com/vercel/turbo/assets/45024986/ae88a14b-05b2-40fe-b168-60d9d345a34d">

All app workspaces dependencies rely on both `esLint` and `tsconfig` workspaces. neither of these workspaces has a `build script`. The "shared" workspace is also depended upon by all apps, and it needs to be built. Currently, Turbo is throwing an error.
<img width="783" alt="image" src="https://github.com/vercel/turbo/assets/45024986/6a60587e-97d4-41aa-9fd9-fe1283010cd5">

**Comment by anthonyshew** [comment]
By default, a package that doesn't implement the script should simply be passed over. This sounds like it has potential to be a bug. Could you open an issue with a reproduction? 🙏

**Comment by mehulkar** [comment, marked as answer]
That error message is saying it didn't find a `build` task in any `turbo.json` files. Check your `turbo.json` in the root of your monorepo for a `build` task. The error message is phrased that way to make it clear that either of those locations could have a task definition, you don't necessarily need it in the root turbo.json if it doesn't apply to all your packages.

If you find that your root `turbo.json` _does_ have a `build` task, please open an issue with the contents of your turbo.json and a reproduction if that's possible!
